code/game/objects/items/mail.dm

```diff
@@ -332,3 +352,172 @@
+	w_class = WEIGHT_CLASS_NORMAL
+	icon = 'icons/obj/device_syndie.dmi'
+	icon_state = "mail_counterfeit_device"
+
+/obj/item/storage/mail_counterfeit_device/Initialize(mapload)
+	. = ..()
+	atom_storage.max_slots = 1
+	atom_storage.allow_big_nesting = TRUE
+	atom_storage.max_specific_storage = WEIGHT_CLASS_NORMAL
+
+/obj/item/storage/mail_counterfeit_device/examine_more(mob/user)
+	. = ..()
+	var/list/msg = list()
+	msg += "[span_notice("<i>You notice the manufacture marking on the side of the device...</i>")]"
+	msg += "\t[span_info("Guerilla Letter Assembler")]"
+	msg += "\t[span_info("GLA Postal Service, right on schedule.")]"
+	return msg
+
+/obj/item/storage/mail_counterfeit_device/attack_self(mob/user, modifiers)
+	var/mail_type = tgui_alert(user, "Is it gonna be an envelope or a normal mail?", "Mail Counterfeiting", list("Mail", "Envelope"))
+	if(isnull(mail_type))
+		return FALSE
+	if(loc != user)
+		return FALSE
+	mail_type = lowertext(mail_type)
+
+	var/mail_armed = tgui_alert(user, "Is it gonna be armed?", "Mail Counterfeiting", list("Yes", "No"))
+	if(isnull(mail_armed))
+		return FALSE
+	if(loc != user)
+		return FALSE
+
+	if(mail_armed == "Yes")
+		mail_armed = TRUE
+	else
+		mail_armed = FALSE
+
+	var/list/mail_recipients = list("*Anyone*")
+	var/list/mail_recipients_input_list = list("*Anyone*")
+	var/list/recipients_dupes = list()
+	for(var/datum/record/locked/person in sort_record(GLOB.manifest.locked))
+		if(isnull(person.mind_ref))
+			continue
+		mail_recipients += person.mind_ref
+
+		// This whole code-hell i had to make just remove iterators from the options names.
+		if(recipients_dupes[person.name] == null)
+			if(mail_recipients_input_list.Find(person.name))
+				recipients_dupes[person.name] = 1
+				mail_recipients_input_list += "[person.name] (1)"
+			else
+				mail_recipients_input_list += "[person.name]"
+		else
+			recipients_dupes[person.name] += 1
+			var/i = recipients_dupes[person.name]
+			mail_recipients_input_list += "[person.name] ([i])"
+		
+	
+	var/recipient = tgui_input_list(user, "Choose a recipient", "Mail Counterfeiting", mail_recipients_input_list)
+	if(isnull(recipient))
+		return FALSE
+	if(!(src in user.contents))
+		return FALSE
+	
+	
+	var/index = mail_recipients_input_list.Find(recipient)
+	
+	var/obj/item/mail/traitor/shady_mail
+	if(mail_type == "mail")
+		shady_mail = new /obj/item/mail/traitor()
+	else
+		shady_mail = new /obj/item/mail/traitor/envelope()
+	
+	shady_mail.made_by_cached_ckey = user.ckey
+	shady_mail.made_by_cached_name = user.mind.name
+
+	if(index == 1)
+		var/mail_name = tgui_input_text(user, "Enter mail title or leave it blank or close the window to get a default one.", "Mail Counterfeiting")
```

Comment:
```suggestion
		var/mail_name = tgui_input_text(user, "Enter mail title, or leave it blank", "Mail Counterfeiting")
```

Keep this terse

Reply:
yep, my sleepy head went a bit funny with grammar and construction of sentences